File: packages/react-native-apk-install/android/src/main/java/com/apkinstall/ApkInstallModule.java
package com.apkinstall;

import androidx.annotation.NonNull;

import android.net.Uri;
import android.os.Build;
import androidx.core.content.FileProvider;
import android.content.Intent;

import com.facebook.react.module.annotations.ReactModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

import java.io.File;
import java.io.IOException;

@ReactModule(name = ApkInstallModule.NAME)
public class ApkInstallModule extends ReactContextBaseJavaModule {
  public static final String NAME = "ApkInstall";
  private ReactApplicationContext _context = null;
  public ApkInstallModule(ReactApplicationContext reactContext) {
    super(reactContext);
    _context = reactContext;
  }

  @Override
  @NonNull
  public String getName() {
    return NAME;
  }


    @ReactMethod
    public void installApk(String path) throws IOException {
      String cmd = "chmod 777 " + path;
      try {
        Runtime.getRuntime().exec(cmd);
      } catch (Exception e) {
        e.printStackTrace();
      }
      File file = new File(path);
      Intent intent = new Intent(Intent.ACTION_VIEW);
      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
      if (Build.VERSION.SDK_INT>= Build.VERSION_CODES.N) {
        Uri apkUri =
          FileProvider.getUriForFile(_context, _context.getApplicationContext().getPackageName() + ".provider", file);
        intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
        intent.setDataAndType(apkUri, "application/vnd.android.package-archive");
      } else {
        intent.setDataAndType(Uri.parse("file://" + path), "application/vnd.android.package-archive");
      }
      _context.startActivity(intent);
    }
}

File: packages/react-native-apk-install/android/src/main/java/com/apkinstall/ApkInstallModule.kt
package com.apkinstall

import android.content.Intent
import android.net.Uri
import android.os.Binder
import android.os.Build
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import java.io.File


class ApkInstallModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
  private var _context: ReactApplicationContext = reactContext

  override fun getName(): String {
    return NAME
  }

  @ReactMethod
  fun installApk(path: String, promise: Promise) {

  }


  companion object {
    const val NAME = "ApkInstall"
  }
}
